controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -319,6 +332,50 @@ func (r *DSCInitializationReconciler) updateStatus(ctx context.Context, original
 	return saved, err
 }
 
+func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
+	if instance.Spec.ServiceMesh.ManagementState == operatorv1.Managed {
+		serviceMeshInitializer := feature.NewFeaturesInitializer(&instance.Spec, servicemesh.ConfigureServiceMeshFeatures)
+
+		if err := serviceMeshInitializer.Prepare(); err != nil {
+			r.Log.Error(err, "failed configuring service mesh resources")
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed configuring service mesh resources")
+
+			return err
+		}
+
+		if err := serviceMeshInitializer.Apply(); err != nil {
+			r.Log.Error(err, "failed applying service mesh resources")
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *DSCInitializationReconciler) configureServerless(instance *dsciv1.DSCInitialization) error {
+	if instance.Spec.Serverless.ManagementState == operatorv1.Managed {
+		serverlessInitializer := feature.NewFeaturesInitializer(&instance.Spec, serverless.ConfigureServerlessFeatures)
+
+		if err := serverlessInitializer.Prepare(); err != nil {
+			r.Log.Error(err, "failed configuring serverless resources")
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed configuring serverless resources")
+
```

Comment:
can we also add `updateStatus() `here to show in the DSCI CR Message field? 
same for `configureServiceMesh()`

Reply:
lets keep it here as-is, and have a dedicated PR to deal with all the updateStatus()  in one go

Reply:
We need a separate issue for that, as we miss this status update but in many other places.

Reply:
ref: https://github.com/opendatahub-io/opendatahub-operator/issues/727